cluster-autoscaler/cloudprovider/clusterapi/clusterapi_controller_test.go

```diff
@@ -346,6 +346,7 @@ func createTestConfigs(specs ...testSpec) []*testConfig {
 								"apiVersion": "infrastructure.cluster.x-k8s.io/v1beta1",
 								"kind":       machineTemplateKind,
 								"name":       "TestMachineTemplate",
+								"namespace":  spec.namespace,
```

Comment:
Using this change (though it doesn't work) as a conversation thread starting point @elmiko 

UT in `cloudprovider/clusterapi/clusterapi_unstructured_test.go` break w/ this change. It appears that the test mocks don't entirely inject the infra template resource, or at least we're getting an error from the lister now that we've moved that list earlier in the execution flow. I tried adding the namespec in here because it seemed to be missing, but that didn't seem to fix the test outcome.

Reply:
To be clear I tested this change on a real cluster running a build of this patch, and scale from zero worked without infra template perms being added to the clusterrole.

Reply:
glad to hear it works live, i don't have a good suggestion on the unit tests at the moment. i'll try to take a look over the next couple days.